pkg/apis/rollouts/v1alpha1/analysis_types.go

```diff
@@ -267,3 +268,16 @@ type ScopeDetail struct {
 	Start  string `json:"start"`
 	End    string `json:"end"`
 }
+
+// I don't like this repetition, but it fits the pattern...
+type WebMetricMetric struct {
+	Url      string            `json:"url"`
+	Headers  []WebMetricHeader `json:"headers,omitempty"`
+	Timeout  int               `json:"timeout,omitempty"`
```

Comment:
I'm debating if the Timeout field should be a string (i.e. 1s vs 1m)

Reply:
I'll leave that one up to you, For all my HTTP work in the past I've always had my timeouts in seconds (rarely in millis), if we are OK with that, then we can be a little more specific with the field name. Otherwise adding in a simple time string parse is fairly trivial. LMKYT